app/src/main/java/com/android/bookswap/ui/map/Map.kt

```diff
@@ -0,0 +1,188 @@
+package com.android.bookswap.ui.map
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.heightIn
+import androidx.compose.foundation.layout.offset
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.itemsIndexed
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.android.bookswap.data.DataBook
+import com.google.android.gms.maps.model.CameraPosition
+import com.google.android.gms.maps.model.LatLng
+import com.google.maps.android.compose.*
+import com.google.maps.android.compose.GoogleMap
+import kotlin.math.roundToInt
+import kotlinx.coroutines.launch
+
+/**
+ * Composable function to display a map with user locations and associated book information.
+ *
+ * This screen renders a GoogleMap that shows books locations as markers. Upon clicking a marker, it
+ * displays a custom info window with the list of books at this location.
+ *
+ * @param listUser List of users (`TempUser`) to display on the map, each containing their location
```

Comment:
In Koltin, for the description, you can use [TempUser] to link a class cleanly (automatically add a link when looking at it in the IDE).